asset-transfer-private-data/application-gateway-go/app.go

```diff
@@ -0,0 +1,358 @@
+	clientOrg1 := newGrpcConnection(
+		tlsCertPathOrg1,
+		peerEndpointOrg1,
+		peerNameOrg1,
+	)
+	defer clientOrg1.Close()
+
+	gatewayOrg1, err := client.Connect(
+		newIdentity(certDirectoryPathOrg1, mspIDOrg1),
+		client.WithSign(newSign(keyDirectoryPathOrg1)),
+		client.WithClientConnection(clientOrg1),
+		client.WithHash(hash.SHA256),
+	)
+	if err != nil {
+		panic(err)
+	}
+	defer gatewayOrg1.Close()
+
+	clientOrg2 := newGrpcConnection(
+		tlsCertPathOrg2,
+		peerEndpointOrg2,
+		peerNameOrg2,
+	)
+	defer clientOrg2.Close()
+
+	gatewayOrg2, err := client.Connect(
+		newIdentity(certDirectoryPathOrg2, mspIDOrg2),
+		client.WithSign(newSign(keyDirectoryPathOrg2)),
+		client.WithClientConnection(clientOrg2),
+		client.WithHash(hash.SHA256),
+	)
+	if err != nil {
+		panic(err)
+	}
+	defer gatewayOrg2.Close()
+
+	// Get the smart contract as an Org1 client.
+	contractOrg1 := gatewayOrg1.GetNetwork(channelName).GetContract(chaincodeName)
+
+	// Get the smart contract as an Org1 client.
+	contractOrg2 := gatewayOrg2.GetNetwork(channelName).GetContract(chaincodeName)
+
+	fmt.Println("~~~~~~~~~~~~~~~~ As Org1 Client ~~~~~~~~~~~~~~~~")
+
+	// Create new assets on the ledger.
+	createAssets(*contractOrg1)
+
+	// Read asset from the Org1's private data collection with ID in the given range.
+	getAssetByRange(*contractOrg1)
+
+	// Attempt to transfer asset without prior approval from Org2, transaction expected to fail.
+	fmt.Println("\nAttempt TransferAsset without prior AgreeToTransfer")
+	err = transferAsset(*contractOrg1, assetID1)
+	if err == nil {
+		doFail("TransferAsset transaction succeeded when it was expected to fail")
+	}
+	fmt.Printf("*** Received expected error: %+v\n", err)
+
+	fmt.Println("\n~~~~~~~~~~~~~~~~ As Org2 Client ~~~~~~~~~~~~~~~~")
+
+	// Read the asset by ID.
+	readAssetByID(*contractOrg2, assetID1)
+
+	// Make agreement to transfer the asset from Org1 to Org2.
+	agreeToTransfer(*contractOrg2, assetID1)
+
+	fmt.Println("\n~~~~~~~~~~~~~~~~ As Org1 Client ~~~~~~~~~~~~~~~~")
+
+	// Read transfer agreement.
+	readTransferAgreement(*contractOrg1, assetID1)
+
+	// Transfer asset to Org2.
+	transferAsset(*contractOrg1, assetID1)
+
+	// Again ReadAsset: results will show that the buyer identity now owns the asset.
+	readAssetByID(*contractOrg1, assetID1)
+
+	// Confirm that transfer removed the private details from the Org1 collection.
+	if org1ReadSuccess := readAssetPrivateDetails(*contractOrg1, assetID1, org1PrivateCollectionName); !org1ReadSuccess {
+		doFail(fmt.Sprintf("Asset private data still exists in %s", org1PrivateCollectionName))
+	}
+
+	fmt.Println("\n~~~~~~~~~~~~~~~~ As Org2 Client ~~~~~~~~~~~~~~~~")
+
+	// Org2 can read asset private details: Org2 is owner, and private details exist in new owner's Collection.
+	if org2ReadSuccess := readAssetPrivateDetails(*contractOrg2, assetID1, org2PrivateCollectionName); !org2ReadSuccess {
+		doFail(fmt.Sprintf("Asset private data not found in %s", org2PrivateCollectionName))
+	}
```

Comment:
The logic is correct on this one but avoid the pointer dereference. I'm not sure whether the extra variable helps or not.

```suggestion
	if !readAssetPrivateDetails(contractOrg2, assetID1, org2PrivateCollectionName) {
		doFail(fmt.Sprintf("Asset private data not found in %s", org2PrivateCollectionName))
	}
```

Reply:
Thanks for the suggestion. Left the variable as explained in previous comment reply.